eth_account/messages.py

```diff
@@ -1,90 +1,232 @@
-from cytoolz import (
-    compose,
+from collections.abc import (
+    Mapping,
 )
-from eth_utils import (
+import json
+from typing import (
+    NamedTuple,
+    Union,
+)
+
+from eth_typing import (
+    Address,
+    Hash32,
+)
+from eth_utils.curried import (
+    ValidationError,
     keccak,
+    text_if_str,
     to_bytes,
+    to_canonical_address,
+    to_text,
 )
 from hexbytes import (
     HexBytes,
 )
 
-from eth_account._utils.signing import (
-    signature_wrapper,
+from eth_account._utils.structured_data.hashing import (
+    hash_domain,
+    hash_message as hash_eip712_message,
+    load_and_validate_structured_message,
+)
+from eth_account._utils.validation import (
+    is_valid_address,
 )
 
+text_to_bytes = text_if_str(to_bytes)
 
-def defunct_hash_message(
-        primitive=None,
+
+# watch for updates to signature format
+class SignableMessage(NamedTuple):
+    """
+    These are the components of an EIP-191_ signable message. Other message formats
+    can be encoded into this format for easy signing. This data structure doesn't need to
+    know about the original message format.
+
+    In typical usage, you should never need to create these by hand. Instead, use
+    one of the available encode_* methods in this module, like:
```

Comment:
Maybe use monospace formatting here?
```suggestion
    one of the available ``encode_*`` methods in this module, like:
```

Reply:
:confused: it doesn't feel quite right. I think it's because it feels wrong to use monospace for anything that's not a proper explicit reference. Since this sort of obliquely references the methods (with the `*`), it doesn't feel right. I could maybe add a bullet list with a couple examples that are referenced explicitly.

Reply:
I guess I had always used single backtick links for explicit references. The double backtick mono-space formatting seems like it's meant for a broader range of situations and might include this case. But whatevs. No biggie.